### PR TITLE
feat(docs-v2): add GuideStepper checkpoint prompt-copy workflow

### PR DESCRIPTION
**Pull request by okane16**
> [!NOTE]
> **Medium Risk**
> Adds new client-side interactive components with persistence and clipboard behavior, and changes the content preprocessing pipeline (regex-based injection + include matching), which could affect MD/MDX rendering if edge cases are missed.
> 
> **Overview**
> Introduces a new interactive `GuideStepper` MDX component for multi-phase guides with nested checkpoints, progress tracking, optional persistence, and a **Copy Prompt** action that concatenates checkpoint markdown into a clipboard-ready prompt.
> 
> Adds a reusable `VerticalProgressSteps` component (numbered/bulleted variants) and wires both components into the MDX renderer exports.
> 
> Updates content preprocessing to run a new `processGuideStepperPrompts` pass (with tests) that injects `rawContent` into `GuideStepper.(Checkpoint|Prompt|AtAGlance)` blocks while skipping code fences, and broadens `INCLUDE_REGEX` to support leading-space include directives (also covered by new tests). Documentation demo MDX is expanded to showcase the new components and includes formatting cleanups.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 67e8b6f505d7ccf8a59f1cc10029d51dc20c52e7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/framework-docs-v2/src/components/mdx-renderer.tsx [hunk 1]
```diff
@@ -81,6 +86,12 @@ const CheckboxGroupWithSubcomponents = Object.assign(CheckboxGroup, {
 const NumberedAccordionWithSubcomponents = Object.assign(NumberedAccordion, {
   Item: NumberedAccordionItem,
 });
+const GuideStepperWithSubcomponents = Object.assign(GuideStepper, {
+  Step: GuideStepperStep,
+  Checkpoint: GuideStepperCheckpoint,
+  AtAGlance: GuideStepperAtAGlance,
+  Prompt: GuideStepperPrompt,
+});
```
[the remark this answers is not in the document]
Comment (okane16) [reply apps/framework-docs-v2/src/components/mdx-renderer.tsx:96]: Checked this against build output after removing the alias. `GuideStepper.AtAGlance` failed to resolve during prerender for `/guides/test-guides/interactive-guide-demo`. Kept `GuideStepperWithSubcomponents = Object.assign(...)` to preserve required runtime subcomponent attachment in MDX rendering. No code change for this thread in 46a788f2f because removing it regresses build behavior.

File: apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+): node is ReactElement<GuideAtAGlanceLikeProps> {
+  if (!isValidElement(node)) return false;
+  const props = node.props as GuideAtAGlanceLikeProps;
+  return (
+    typeof props.rawContent === "string" &&
+    typeof props.id === "undefined" &&
+    (typeof props.title === "undefined" || typeof props.title === "string") &&
+    typeof props.number === "undefined"
+  );
+}
```
[the remark this answers is not in the document]
Comment (okane16) [reply apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts]: Fixed in 46a788f2f. Added an invariant comment above `isGuideStepperCheckpointElement` documenting that checkpoints intentionally rely on `props.number === undefined` and that a discriminant prop can replace this later.

File: apps/framework-docs-v2/src/lib/includes.test.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  it("matches include directives with leading indentation", () => {
+    INCLUDE_REGEX.lastIndex = 0;
+    const content = "    :::include /shared/guides/foo.mdx";
+    const matches = [...content.matchAll(INCLUDE_REGEX)];
+
+    expect(matches).toHaveLength(1);
+    expect(matches[0]?.[1]).toBe("/shared/guides/foo.mdx");
+  });
+});
```
[the remark this answers is not in the document]
Comment (okane16) [reply apps/framework-docs-v2/src/lib/includes.test.ts:51]: Fixed in 46a788f2f. Added edge-case tests for no-path directives, partial keyword matches, and tab-indented directives. To make those tests meaningful, tightened `INCLUDE_REGEX` to require space-indented directives with a non-empty path.

File: apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper.tsx [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+  sanitizeCompletedStepIds,
+} from "./guide-stepper-utils";
+
+interface GuideStepperProps {
+  id?: string;
+  persist?: boolean;
+  defaultExpanded?: string[];
+  className?: string;
+  children: ReactNode;
+}
```
[the remark this answers is not in the document]
Comment (okane16) [reply apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper.tsx]: Fixed in 46a788f2f. Exported `GuideStepperProps` directly as `export interface` and removed the redundant `export type { GuideStepperProps }` re-export at the bottom of the file.

File: apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  id?: string;
+  title: string;
+  children: ReactNode;
+}
+
+function VerticalProgressStepItemComponent({
+  children,
+}: VerticalProgressStepItemProps) {
+  return <>{children}</>;
+}
```
[the remark this answers is not in the document]
Comment (okane16) [reply apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps.tsx:58]: Fixed in 46a788f2f. Added an inline comment above `VerticalProgressStepItemComponent` clarifying that `id` and `title` are intentionally consumed by the parent via `item.props`.

File: apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+function VerticalProgressStepsRoot({
+  variant = "numbered",
+  className,
+  children,
+}: VerticalProgressStepsProps) {
+  const resolvedVariant = getProgressStepsVariant(variant);
+  const ListTag = resolvedVariant === "bulleted" ? "ul" : "ol";
+  const items = Children.toArray(children).filter(
+    isVerticalProgressStepItemElement,
+  ) as ReactElement<VerticalProgressStepItemProps>[];
```
[the remark this answers is not in the document]
Comment (okane16) [reply apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps.tsx:70]: Kept as-is intentionally. `getProgressStepsVariant` is retained as runtime normalization for MDX-authored props, so unsupported variant strings still fall back safely to `"numbered"` instead of assuming type-level guarantees at runtime.

File: apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps-utils.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { type ReactElement, type ReactNode, isValidElement } from "react";
+
+export type ProgressStepsVariant = "numbered" | "bulleted";
+
+interface VerticalProgressStepLikeProps {
+  id?: unknown;
```
Comment (callicles) [inline apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps-utils.ts]: Why is that type unknown?

File: apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps-utils.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  title?: unknown;
```
Comment (callicles) [inline apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps-utils.ts]: same here

File: apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+  defaultExpanded?: string[];
+}
+
+interface CompletionToggleInput {
+  stepId: string;
+  checked: boolean;
+}
+
+interface GuideStepLikeProps {
+  id?: unknown;
```
Comment (callicles) [inline apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts]: There is no reason to use unknown here

File: apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+  number?: unknown;
+  rawContent?: unknown;
+}
+
+interface GuideAtAGlanceLikeProps {
+  rawContent?: unknown;
+  id?: unknown;
+  title?: unknown;
+  number?: unknown;
+}
```
[the remark this answers is not in the document]
Comment (callicles) [reply apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts]: This is a valid comment

**Review by callicles — COMMENTED** [review]
Some more polish to do

File: apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+): node is ReactElement<GuideAtAGlanceLikeProps> {
+  if (!isValidElement(node)) return false;
+  const props = node.props as GuideAtAGlanceLikeProps;
+  return (
+    typeof props.rawContent === "string" &&
+    typeof props.id === "undefined" &&
+    (typeof props.title === "undefined" || typeof props.title === "string") &&
+    typeof props.number === "undefined"
+  );
+}
```
[the remark this answers is not in the document]
Comment (callicles) [reply apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts]: This is a good point - @okane16 you should add a _type field there to address this

File: apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps-utils.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+export function getProgressStepsVariant(value?: string): ProgressStepsVariant {
+  return value === "bulleted" ? "bulleted" : "numbered";
+}
+
+export function isVerticalProgressStepItemElement(
+  node: ReactNode,
+): node is ReactElement<VerticalProgressStepLikeProps> {
+  if (!isValidElement(node)) return false;
+  const props = node.props as VerticalProgressStepLikeProps;
+  return (
```
Comment (callicles) [inline apps/framework-docs-v2/src/components/mdx/interactive/vertical-progress-steps-utils.ts]: The test here feels very weak as well. We should have a `_type` field to distinguish

File: apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (okane16) [reply apps/framework-docs-v2/src/components/mdx/interactive/guide-stepper-utils.ts]: should have addressed this in latest push
